### ISSUE TITLE
[On Hold #44778] Submit Expense - Inconsistent Error Messages for Password Protected PDFs on Mweb and Android

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#44091]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.86-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** abebemiherat@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. On Android: Sign in with Expensify > FAB > Submit Expense
2. Add an amount and attach a password-protected PDF
3. On Mweb: Repeat the above steps
4. Observe the error messages

## Expected Result:
The error message should be consistent for the same password-protected PDF across both platforms

## Actual Result:
Different error messages appear for password-protected PDFs on Mweb and Android

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/6106ce5a-5d16-4611-8b18-a9f646403ade

![9!1](https://github.com/Expensify/App/assets/78819774/9ef00396-ac70-4894-9221-93ad5fad34a5)

![19!2](https://github.com/Expensify/App/assets/78819774/1f59e32e-c0f7-4256-b1e9-1d0b0a77696a)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a71dc36f6e73ac8e</li>
        <li>Upwork Job ID: 1806742272201940772</li>
        <li>Last Price Increase: 2024-07-05</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @s77rt</details>

**Comment by lanitochka17** [comment]
@CortneyOfstad  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by etCoderDysto** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Inconsistent Error Messages for Password Protected PDFs on Mweb and Android
### What is the root cause of that problem?
The root cause is that we are handling errors concerning password protected pdf files for native apps [here](https://github.com/Expensify/App/blob/f618a94c486b80aebab584c52288a7013ffad64a/src/components/MoneyRequestConfirmationList.tsx#L1237-L1238). That is because password protected pdf error is not caught by the validation put [here](https://github.com/Expensify/App/blob/f618a94c486b80aebab584c52288a7013ffad64a/src/pages/iou/request/step/IOURequestStepScan/index.native.tsx#L164-L184) for native devices. Hence 'Attachment error' message is displayed because of the default error key `attachmentPicker.attachmentError`.
https://github.com/Expensify/App/blob/f618a94c486b80aebab584c52288a7013ffad64a/src/components/MoneyRequestConfirmationList.tsx#L1237-L1238

For web, we handle password protected pdf here
https://github.com/Expensify/App/blob/f618a94c486b80aebab584c52288a7013ffad64a/src/pages/iou/request/step/IOURequestStepScan/index.tsx#L216-L219
### What changes do you think we should make in order to solve the problem?
Change web error message key to `attachmentPicker.attachmentError` here
https://github.com/Expensify/App/blob/f618a94c486b80aebab584c52288a7013ffad64a/src/pages/iou/request/step/IOURequestStepScan/index.tsx#L216-L219

### What alternative solutions did you explore? (Optional)

Change native platform error message to `attachmentPicker.wrongFileType` here
https://github.com/Expensify/App/blob/f618a94c486b80aebab584c52288a7013ffad64a/src/components/MoneyRequestConfirmationList.tsx#L1237-L1238

**Comment by CortneyOfstad** [comment]
Sorry was OoO!

I think this may be more BE (tied to design) so going to get confirmation!

**Comment by CortneyOfstad** [comment]
Asked [here](https://expensify.enterprise.slack.com/archives/C03TME82F/p1719341984264519)

**Comment by CortneyOfstad** [comment]
Bumped the thread in design again 👍

**Comment by CortneyOfstad** [comment]
Alright, confirmed with the Design Team that we want `Invalid file type` to be the main error across the board!

**Comment by s77rt** [comment]
@Krishna2323 @etCoderDysto @cretadn22 Thank you for the proposals. The inconsistency we are having concerns both the error message being different as well as the timing of the verification phase (scan vs confirmation page). Can we add support to pdf verification on the scan step on native to be consist with web and then remove the one in the confirmation page?

**Comment by CortneyOfstad** [comment]
@Krishna2323 @etCoderDysto @cretadn22 bumps on the above ^^^^ thank you!

**Comment by s77rt** [comment]
@CortneyOfstad I'm on it

**Comment by s77rt** [comment]
@etCoderDysto The `pdfjsLib` does not support rendering on native but I would assume that reading just a buffer of bytes to tell if the file is password protected or not should work fine. Have you tried reading the document?

**Comment by etCoderDysto** [comment]
> @etCoderDysto The `pdfjsLib` does not support rendering on native but I would assume that reading just a buffer of bytes to tell if the file is password protected or not should work fine. Have you tried reading the document?

I haven't tried that. I will try it and get back to you!

**Comment by s77rt** [comment]
Still waiting on proposals

**Comment by s77rt** [comment]
@cretadn22 Your proposal will only change the text error but still keep the inconsistency where on web the validation is done on the scan page while on native it's done on the confirmation page.

I would prefer doing one of the following:
1. Make the pdfjs library work on native (only for checking if the pdf is password protected and not rendering)
2. Extend react-native-pdf as you mentioned

**Comment by s77rt** [comment]
@cretadn22 

> Upon further investigation into react-native-pdf, I found that it doesn't offer a direct API specifically for checking password protection.

There is an exported [`PdfManager`](https://github.com/wonday/react-native-pdf/blob/master/PdfManager.js) that we can use to load the file without rendering.

**Comment by s77rt** [comment]
@CortneyOfstad Let hold this on https://github.com/Expensify/App/pull/44778

**Comment by CortneyOfstad** [comment]
Sounds good — thanks @s77rt!

**Comment by CortneyOfstad** [comment]
Deployed to production! Going to close 🎉
